arm/Microsoft.Authorization/roleAssignments/.bicep/nested_rbac_rg.bicep

```diff
@@ -316,13 +324,13 @@ var builtInRoleNames_var = {
 
 var roleDefinitionId_var = (contains(builtInRoleNames_var, roleDefinitionIdOrName) ? builtInRoleNames_var[roleDefinitionIdOrName] : roleDefinitionIdOrName)
 
-resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2021-04-01-preview' = {
   name: guid(subscriptionId, resourceGroupName, roleDefinitionId_var, principalId)
   properties: {
     roleDefinitionId: roleDefinitionId_var
     principalId: principalId
     description: !empty(description) ? description : null
-    principalType: !empty(principalType) ? principalType : null
+    principalType: any(!empty(principalType) ? principalType : null)
```

Comment:
Are you sure the `any` is needed here? Linter does not light up in my case: 
![image](https://user-images.githubusercontent.com/5365358/144725222-9894de45-1f87-4c9a-a8f4-10c6f86f0c62.png)

Reply:
Have you switched the API to a currently known one by bicep? latest is `2020-08-01-preview`

![image](https://user-images.githubusercontent.com/28486158/144725318-c3ff9c41-87c6-44ed-93a8-9a3daa97f636.png)

I'm running on version `0.4.1.008`

Reply:
Just checked your branch out

Reply:
Here is the message:
```
The property "principalType" expected a value of type "'ForeignGroup' | 'Group' | 'ServicePrincipal' | 'User' | null" but the provided value is of type "'' | 'Device' | 'ForeignGroup' | 'Group' | 'ServicePrincipal' | 'User' | null".bicep(BCP036)
```

I'd wrap the `principalType` as it is the one containing `''` as linter indicates that `null` is supported. Thoughts?
```bicep
principalType: !empty(principalType) ? any(principalType) : null
```

Reply:
@MrMCake it is now like this `any(principalType)`. Tested again against the dependency pipeline and the module parameters. All working